recursion/problem-set-recursion-pt-1.md

```diff
@@ -0,0 +1,277 @@
+
+Here are the tests:
+
+```python
+def test_reverse_word():
+    assert reverse("cat") == "tac"
+
+
+def test_reverse_single_character():
+    assert reverse("a") == "a"
+
+
+def test_reverse_empty_string():
+    assert reverse("") == ""
+```
+
+### !end-question
+### !tests
+```python
+import unittest
+from main import *
+
+class TestChallenge(unittest.TestCase):
+    def test_reverse_word(self):
+        self.assertEqual(reverse("cat"), "tac")
+
+    def test_reverse_single_character(self):
+        self.assertEqual(reverse("a"), "a")
+
+    def test_reverse_empty_string(self):
+        self.assertEqual(reverse(""), "")
+```
+### !end-tests
+### !explanation
+
+An example of a working implementation:
+
+```python
+def reverse(text):
+    if len(text) <= 1:
+        return text
+    # Use Python string slicing to get the
+    # first character of text and the
+    # remaining characters
+    first_char = text[:1]
+    remaining_chars = text[1:]
+    return reverse(remaining_chars) + first_char
+```
+
+### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 3 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: i99SZO
+* title: Recursion
+* points: 3
+### !question
+
+Write a function `bunny` that accepts an integer parameter `count`. `count` represents a number of bunnies and each bunny has two big floppy ears. We want to compute the total number of ears across all the bunnies recursively (without loops or multiplication).
+
+Here are the tests:
+
+```python
+def test_bunny_zero_bunnies():
+    assert bunny(0) == 0
+
+
+def test_bunny_one_bunny_has_two_ears():
+    assert bunny(1) == 2
+
+
+def test_bunny_fifty_bunnies_have_100_ears():
+    assert bunny(50) == 100
+```
+
+### !end-question
+### !tests
+```python
+import unittest
+from main import *
+
+class TestChallenge(unittest.TestCase):
+    def test_bunny_zero_bunnies(self):
+        self.assertEqual(bunny(0), 0)
+
+    def test_bunny_one_bunny_has_two_ears(self):
+        self.assertEqual(bunny(1), 2)
+
+    def test_bunny_fifty_bunnies_have_100_ears(self):
+        self.assertEqual(bunny(50), 100)
+```
+### !end-tests
+### !explanation
+
+An example of a working implementation:
+
+```python
+def bunny(count):
+    if not count:
+        return count
+    else:
+        return 2 + bunny(count-1)
+```
+
+### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 4 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: hhiNnl
+* title: Recursion
+* points: 3
+### !question
+
+Write a function `is_nested_parens` that accepts a string `input` of only parenthesis. It returns `True` if those parentheses are properly nested, and `False` if they are not. You may assume that no non-parenthesis characters will be passed to this function.
```

Comment:
throughout this problem, I suggest replacing input with parens to avoid using the built-in input as a variable
```suggestion
Write a function `is_nested_parens` that accepts a string `parens` of only parentheses. It returns `True` if those parentheses are properly nested, and `False` if they are not. You may assume that no non-parenthesis characters will be passed to this function.
```